code/datums/components/fishing_spot.dm

```diff
@@ -28,6 +33,43 @@
 		return COMPONENT_NO_AFTERATTACK
 	return NONE
 
+///If the fish source has fishes that are shown in the
+/datum/component/fishing_spot/proc/on_examined(datum/source, mob/user, list/examine_text)
+	SIGNAL_HANDLER
+	if(!HAS_MIND_TRAIT(user, TRAIT_EXAMINE_FISHING_SPOT))
+		return
+
+	var/has_known_fishes = FALSE
+	for(var/reward in fish_source.fish_counts)
+		if(!ispath(reward, /obj/item/fish))
+			continue
+		var/obj/item/fish/prototype = reward
+		if(initial(prototype.show_in_catalog))
+			has_known_fishes = TRUE
+			break
+	if(!has_known_fishes)
+		return
+
+	. += span_tinynoticeital("This is a fishing spot. You can look again to list its fishes...")
```

Comment:
```suggestion
	examine_text += span_tinynoticeital("This is a fishing spot. You can look again to list its fishes...")
```

Remember, `SIGNAL_HANDLER` procs (as in procs that we register signals to) should only return bitfields.